adds default system prompt

rperez030 opened the pull request:
Adds a default system prompt that will be used if the setting to remember the content of the system field is not selected or if the user has not defined a previous prompt. Closes #29. I could not test this thoroughly because I had trouble building the add-on, but the change is small enough. Please review before merging.

Nael-Sayegh commented:
Hello,
Very good PR idea.
I think we should add an option to disable the default system text if the user does not want to use it.
The best would be to add it in the settings panel.
What do you think?

rperez030 commented:
what about using the default prompt only if the setting to remember the content of the system field is unchecked? That would provide a mechanism to disable the default prompt without adding an additional setting. We could also add a context menu option to reset the system prompt to default if the user wants it back without unchecking the setting. thoughts?

Nael-Sayegh commented:
The problem with the "remember system field" checkbox is that it does not allow saving an empty system field.
This behavior should be changed so that everything works correctly with your new option.

rperez030 commented:
I understand now. I think allowing an empty field is a more intuitive approach, but happy to proceed however you think is best. I can change the logic to allow an empty field, or add a new setting.

Nael-Sayegh commented:
I think the best solution is to allow an empty field; I agree with you that adding a new parameter will only make the interface more cumbersome when this could be avoided.

rperez030 commented:
I made a few changes. It is now possible to save an empty field as the last system prompt. If the system is unchecked, it will revert to the default system prompt. I also noticed that the system prompt was not being removed even if the user unchecks the setting, so that it is still remembered when the setting is checked again, even though the saved system prompt is no longer the last one. I made sure that the system key is removed from self.data if the setting is unchecked. Thoughts?

rperez030 commented:
I made the following changes:

* Added a custom context menu with the option to reset system field to default. I don't have experience with wxPython and couldn't''t find an elegant way to extend the OS context menu, so added the most common options manually. I would be happy to improve it if someone knows a better way to do that.
* Moved the definition of default system prompt to consts,py and addressed format and spelling as suggested by @aaclause.
* Changed the save system setting to be on by default as suggested by @aaclause. It makes more sense to me as well to have it on by default.

Thank you both @Nael-Sayegh  and @aaclause  for working with me on this.

rperez030 commented:
Thanks for the resources. I think I understood what you meant now.

  I also added an `addStandardMenuOptions ` method so that it can be reused in any other field to add the standard commands.
